From 0d4022da784b1fc29056ef127e00e952fdc65f1b Mon Sep 17 00:00:00 2001
From: Dominik Holland <dominik.holland@pelagicore.com>
Date: Fri, 20 Dec 2013 09:13:41 +0100
Subject: [PATCH] Allow to build only dbus tools

On a target build the only suitable tools are the dbus tools
---
 examples/examples.pro | 2 +-
 src/src.pro           | 8 ++++----
 tests/tests.pro       | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/examples/examples.pro b/examples/examples.pro
index 4955969..cd0f0b4 100644
--- a/examples/examples.pro
+++ b/examples/examples.pro
@@ -1,3 +1,3 @@
 TEMPLATE = subdirs
-qtHaveModule(widgets): SUBDIRS += help designer linguist uitools assistant
+!dbusonly:qtHaveModule(widgets): SUBDIRS += help designer linguist uitools assistant
 
diff --git a/src/src.pro b/src/src.pro
index c8756db..55287f1 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -1,6 +1,6 @@
 TEMPLATE = subdirs
 
-qtHaveModule(widgets) {
+!dbusonly:qtHaveModule(widgets) {
     no-png {
         message("Some graphics-related tools are unavailable without PNG support")
     } else {
@@ -14,15 +14,15 @@ qtHaveModule(widgets) {
     }
 }
 
-SUBDIRS += linguist
+!dbusonly:SUBDIRS += linguist
 
-mac {
+!dbusonly:mac {
     SUBDIRS += macdeployqt
 }
 
 qtHaveModule(dbus): SUBDIRS += qdbus
 
-qtNomakeTools( \
+!dbusonly:qtNomakeTools( \
     pixeltool \
     qtconfig \
     macdeployqt \
diff --git a/tests/tests.pro b/tests/tests.pro
index 85e4f3a..1637ecf 100644
--- a/tests/tests.pro
+++ b/tests/tests.pro
@@ -1,2 +1,2 @@
 TEMPLATE = subdirs
-SUBDIRS +=  auto
+!dbusonly:SUBDIRS +=  auto
-- 
1.8.5

